lib/src/flare_animation.dart

```diff
@@ -0,0 +1,213 @@
+    this.snapToEnd = false,
+    this.controller,
+    this.callback,
+    this.color,
+    this.shouldClip = true,
+    this.sizeFromArtboard = false,
+    this.artboard,
+    this.useAntialias = true,
+  }) : filename = null;
+
+  FlareActorRenderObject _renderObject;
+
+  // Flare only allows the render box to be loaded if it is considered "attached", we need this ugly dumb thing here to do that.
+  final _pipelineOwner = _FlareActorComponentPipelineOwner();
+
+  /// The animation size base.
+  double width = 0.0, height = 0.0;
+
+  // Fields are ported from flare actor widget
+  /// Mirror to [FlareActor.filename]
+  final String filename;
+
+  /// Mirror to [FlareActor.flareProvider]
+  final AssetProvider flareProvider;
+
+  /// Mirror to [FlareActor.artboard]
+  final String artboard;
+
+  /// Mirror to [FlareActor.animation]
+  final String animation;
+
+  /// Mirror to [FlareActor.snapToEnd]
+  final bool snapToEnd;
+
+  /// Mirror to [FlareActor.fit]
+  final BoxFit fit;
+
+  /// Mirror to [FlareActor.alignment]
+  final Alignment alignment;
+
+  /// Mirror to [FlareActor.isPaused]
+  final bool isPaused;
+
+  /// Mirror to [FlareActor.shouldClip]
+  final bool shouldClip;
+
+  /// Mirror to [FlareActor.controller]
+  final FlareController controller;
+
+  /// Mirror to [FlareActor.callback]
+  final FlareCompletedCallback callback;
+
+  /// Mirror to [FlareActor.color]
+  final Color color;
+
+  /// Mirror to [FlareActor.boundsNode]
+  final String boundsNode;
+
+  /// Mirror to [FlareActor.sizeFromArtboard]
+  final bool sizeFromArtboard;
+
+  /// When false disables antialiasing on drawables.
+  final bool useAntialias;
+
+  void init() {
+    _renderObject = FlareActorRenderObject()
+      ..assetProvider =
+          flareProvider ?? AssetFlare(bundle: Flame.bundle, name: filename)
+      ..alignment = alignment
+      ..animationName = animation
+      ..snapToEnd = snapToEnd
+      ..isPaused = isPaused
+      ..controller = controller
+      ..completed = callback
+      ..color = color
+      ..shouldClip = shouldClip
+      ..boundsNodeName = boundsNode
+      ..useIntrinsicSize = sizeFromArtboard
+      ..artboardName = artboard
+      ..useAntialias = useAntialias;
+
+    _loadRenderBox();
+  }
+
+  void render(Canvas canvas, Offset position) {
+    assert(_renderObject != null,
+        "FlareActorAnimation was rendered before initialization. Run FlareActorAnimation.init() before rendering it");
+    final bounds = _renderObject.aabb;
+    if (bounds != null) {
+      _paintActor(canvas, bounds, position);
+    }
+  }
+
+  void advance(double dt) {
+    assert(_renderObject != null,
+        "FlareActorAnimation was advanced before initialization. Run FlareActorAnimation.init() before calling .advance");
+    _renderObject.advance(dt);
+  }
+
+  void destroy() {
+    assert(_renderObject != null,
+        "FlareActorAnimation was destroyed before initialization. Run FlareActorAnimation.init() before destroying it");
```

Comment:
Minor suggestion, but this could be a method, so we don't repeat the full message on three places, and the method could receive only the details.